crates/sui-storage/src/object_store/mod.rs

```diff
@@ -91,6 +95,9 @@ pub struct ObjectStoreConfig {
     #[serde(default = "default_object_store_connection_limit")]
     #[arg(long, default_value_t = 20)]
     pub object_store_connection_limit: usize,
+    #[serde(default)]
+    #[arg(long, default_value_t = false)]
+    pub no_sign_request: bool,
```

Comment:
I am open to making this default to true in `sui-tool`. The only reason i didnt do it was we dont have no-sign-request ready for db snapshots yet. But it is also coming soon

Reply:
Ah I see. But in that case the config flag would get ignored, so whether it defaults to true or false doesn't matter. And since the user is not being explicit about the config value in that case, there is also no chance for confusion (unless they are manually checking out the fully generated config somehow)

Reply:
I think defaulting to false maintains the current expectations/communication on our end. We won't enable requester pays until we're in a solid place with R2 and have well communicated the change; at that point we can flip the flag if there are complaints?